### PR TITLE
Shell unify commands ret val

### PR DESCRIPTION
**Pull request by jakub-uC**
This PR is first step to implement CI tests shell module.
I have unified usage of a function shell_cmd_precheck, now it returns error messge, not only true/false.

Return values have now follwoing meaning:
-  0		          if check passed
- 1                      if help was requested
-  -EINVAL	  if wrong argument count

### CODE REVIEW

File: include/shell/shell.h [hunk 1]
```diff
@@ -572,12 +572,14 @@ int shell_prompt_change(const struct shell *shell, char *prompt);
  * @param[in] opt	  Pointer to the optional option array.
```
[the remark this answers is not in the document]
Comment (jarz-nordic) [reply include/shell/shell.h]: I was trying to achieve the same effect as in a PR description (bullet points).
`-EINVAL` is one of 3 possible return parameters hence it is in a new line.

**Review by nordic-krch — APPROVED** [review]
lgtm

File: include/shell/shell.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (jukkar) [reply include/shell/shell.h]: Ah ok, I clearly missed the point :)

**Review by jukkar — APPROVED** [review]
LGTM

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@dbkinder skipping your review since the changes to doc are very minor and they don't affect doc text, only code.

**Pull request merged**